src/integration-test/integration/MappingsClient.test.ts

```diff
@@ -0,0 +1,578 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import * as chaiAsPromised from "chai-as-promised";
+import { expect, use } from "chai";
+import { CalculatedPropertyCreate, CalculatedPropertyType, CalculatedPropertyUpdate, CustomCalculationCreate, CustomCalculationUpdate, DataType, ECProperty, GroupCreate, GroupPropertyCreate, GroupPropertyUpdate, GroupUpdate, MappingCopy, MappingCreate, MappingsClient, MappingUpdate, QuantityType } from "../../reporting";
+import "reflect-metadata";
+import { accessToken, testIModel, testIModelGroup } from "../utils/";
+use(chaiAsPromised);
+
+describe("Mapping Client", () => {
+  const mappingsClient: MappingsClient = new MappingsClient();
+
+  const mappingIds: Array<string> = [];
+  let groupId: string;
+  let groupPropertyId: string;
+  let calculatedPropertyId: string;
+  let customCalculationId: string;
+
+  before(async () => {
+    // create mappings
+    const newMapping: MappingCreate = {
+      mappingName: "Test1",
+    };
+    let mapping = await mappingsClient.createMapping(accessToken, testIModel.id, newMapping);
+    expect(mapping).to.not.be.undefined;
+    expect(mapping.mappingName).to.be.eq("Test1");
```

Comment:
I think it would be best if you would remove pretty much all of the `expect`ations from the `before` step, because it duplicates the same assertions in individual tests. That way there will be a lot less code duplication.

Reply:
Removed all expects from the before section of tests except for the one that checks if an extraction has been completed successfully in OData tests